src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -233,6 +236,32 @@ public static bool TryParse(string value, bool allowFloating, out VersionRange v
                 }
             }
 
+            // Illogical version range detection
+            if (minVersion != null && maxVersion != null
+                && minVersion >= maxVersion
+                && !(minVersion == maxVersion && isMinInclusive && isMaxInclusive))  // Exclude (9.0.0,9.0.0) since it's used as empty version range.
+            {
+                if (partsLength == 1)
```

Comment:
This check can still be a lot simpler. 

All comments aside, I'd expect `return false` covers all scenarios here and you don't need any of the parts checks.

Reply:
Ok. I addressed this one. Please check.